pkg/chunk/repair_test.go

```diff
@@ -0,0 +1,296 @@
+			t.Fatal("receipt not received")
+		}
+
+		if !receipt.Address.Equal(c1.Address()) {
+			t.Fatalf("invalid address in receipt: expected %s received %s", c1.Address(), receipt.Address)
+		}
+
+	})
+
+	t.Run("repair-chunk-not-present", func(t *testing.T) {
+		// generate test chunk, store and publisher
+		c2 := chunktesting.GenerateTestRandomChunk()
+
+		// create a mock storer
+		mockStorer := storemock.NewStorer()
+		defer mockStorer.Close()
+
+		// create a mock pushsync service
+		pushServiceCalled := false
+		pushSyncService := pushsyncmock.New(func(ctx context.Context, chunk swarm.Chunk) (*pushsync.Receipt, error) {
+			pushServiceCalled = true
+			return nil, nil
+		})
+
+		// create the chunk repair handler
+		repairHandler := chunk.NewRepairHandler(mockStorer, logger, pushSyncService)
+
+		//create a trojan message to trigger the repair of the chunk
+		testTopic := trojan.NewTopic("foo")
+		maxPayload := make([]byte, swarm.SectionSize)
+		var msg trojan.Message
+		copy(maxPayload, c2.Address().Bytes())
+		msg, err := trojan.NewMessage(testTopic, maxPayload)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// invoke the chunk repair handler
+		repairHandler(msg)
+
+		if pushServiceCalled {
+			t.Fatal("push service called even if the chunk is not present")
+		}
+	})
+
+	t.Run("repair-chunk-closest-peer-not-present", func(t *testing.T) {
+		// generate test chunk, store and publisher
+		c3 := chunktesting.GenerateTestRandomChunk()
+
+		// create a mock storer
+		mockStorer := storemock.NewStorer()
+		defer mockStorer.Close()
+		_, err := mockStorer.Put(context.Background(), storage.ModePutRequest, c3)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// create a mock pushsync service
+		var receiptError error
+		pushSyncService := pushsyncmock.New(func(ctx context.Context, chunk swarm.Chunk) (*pushsync.Receipt, error) {
+			receiptError = errors.New("invalid receipt")
+			return nil, receiptError
+		})
+
+		// create the chunk repair handler
+		repairHandler := chunk.NewRepairHandler(mockStorer, logger, pushSyncService)
+
+		//create a trojan message to trigger the repair of the chunk
+		testTopic := trojan.NewTopic("foo")
+		maxPayload := make([]byte, swarm.SectionSize)
+		var msg trojan.Message
+		copy(maxPayload, c3.Address().Bytes())
+		msg, err = trojan.NewMessage(testTopic, maxPayload)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// invoke the chunk repair handler
+		repairHandler(msg)
+
+		if receiptError == nil {
+			t.Fatal("pushsync did not generate a receipt error")
+		}
+	})
+}
+
+// newTestNetStore creates a test store with a set RemoteGet func
+func newTestNetStore(t *testing.T, recoveryFunc chunk.RecoveryHook) storage.Storer {
+	t.Helper()
+	storer := mock.NewStorer()
+	logger := logging.New(ioutil.Discard, 5)
+
+	mockStorer := storemock.NewStorer()
+	serverMockAccounting := accountingmock.NewAccounting()
+	price := uint64(10)
+	pricerMock := accountingmock.NewPricer(price, price)
+	peerID := swarm.MustParseHexAddress("9ee7add7")
```

Comment:
could you please explain the reason behind the `price` and `peerID` values?

Reply:
no reason. They just need to be there for the retreival to work without panic. I changed the values so they they clearly look dummy.

Reply:
thanks